Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccreissuance/core/reissuer/DccReissuerTest.kt

```diff
@@ -0,0 +1,182 @@
+            ),
+            subtitleText = SingleText(
+                type = "string",
+                localizedText = mapOf("de" to "Text"),
+                parameters = listOf()
+            ),
+            longText = SingleText(
+                type = "string",
+                localizedText = mapOf("de" to "Langer Text"),
+                parameters = listOf()
+            ),
+            faqAnchor = "dcc_admission_state"
+        ),
+        certificateToReissue = Certificate(
+            certificateRef = CertificateRef(
+                barcodeData = "HC1:6BFOXN...",
+            )
+        ),
+        accompanyingCertificates = listOf(
+            Certificate(
+                certificateRef = CertificateRef(
+                    barcodeData = "HC1:6BFOXN..."
+                )
+            )
+        )
+    )
+
+    @BeforeEach
+    fun setUp() {
+        MockKAnnotations.init(this)
+
+        coEvery { dccSwapper.swap(any(), any()) } just Runs
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = emptyList()
+        )
+    }
+
+    @Test
+    fun `startReissuance throws DCC_RI_NO_RELATION if no reissuances`() = runBlockingTest {
+        shouldThrow<DccReissuanceException> {
+            dccReissuer().startReissuance(dccReissuanceDescriptor = certificateReissuance)
+        }.errorCode shouldBe DccReissuanceException.ErrorCode.DCC_RI_NO_RELATION
+
+        coVerify(exactly = 0) { dccSwapper.swap(any(), any()) }
+    }
+
+    @Test
+    fun `startReissuance throws DCC_RI_NO_RELATION if no relation index`() = runBlockingTest {
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = listOf(
+                DccReissuanceResponse.DccReissuance(
+                    certificate = "HC1:6BFOXN...",
+                    relations = listOf(
+                        DccReissuanceResponse.Relation(
+                            index = 1,
+                            action = "replace"
+                        )
+                    )
+                )
+            )
+        )
+        shouldThrow<DccReissuanceException> {
+            dccReissuer().startReissuance(dccReissuanceDescriptor = certificateReissuance)
+        }.errorCode shouldBe DccReissuanceException.ErrorCode.DCC_RI_NO_RELATION
+
+        coVerify(exactly = 0) { dccSwapper.swap(any(), any()) }
+    }
+
+    @Test
+    fun `startReissuance throws DCC_RI_NO_RELATION if no relation action`() = runBlockingTest {
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = listOf(
+                DccReissuanceResponse.DccReissuance(
+                    certificate = "HC1:6BFOXN...",
+                    relations = listOf(
+                        DccReissuanceResponse.Relation(
+                            index = 1,
```

Comment:
Use `0` as index here so that we assure that the invalid action and not the invalid index is causing the exception?